engine/src/hopeit/server/web.py

```diff
@@ -173,6 +174,9 @@ async def stream_startup_hook(app_config: AppConfig, *args, **kwargs):
     """
     app_engine = runtime.server.app_engines[app_config.app_key()]
     for event_name, event_info in app_engine.effective_events.items():
```

Comment:
This iterates over `effective_events`. We must ensure that the right `group` value is copied when creating the effective events list that contains event split with a SHUFFLE keyword (I think in events.py or steps.py).